source/Octopus.Client/Repositories/Async/TaskRepository.cs

```diff
@@ -142,7 +142,7 @@ public async Task<TaskResource> ExecuteTentacleUpgrade(string description = null
             return await Create(resource).ConfigureAwait(false);
         }
 
-        public async Task<TaskResource> ExecuteAdHocScript(string scriptBody, string[] machineIds = null, string[] environmentIds = null, string[] targetRoles = null, string description = null, string syntax = "PowerShell")
+        public async Task<TaskResource> ExecuteAdHocScript(string scriptBody, string[] machineIds = null, string[] environmentIds = null, string[] targetRoles = null, string description = null, string syntax = "PowerShell", BuiltInTasks.AdHocScript.ScriptTaskTarget? scriptTaskTarget = null)
```

Comment:
Food for thought: is using an enum here going to byte us with backwards compat? Should we just use a string instead?